src/hyper_rustls_adapter.rs

```diff
@@ -136,6 +136,11 @@ impl Adapter for HyperRustlsAdapter {
             .header(ACCEPT, header::APPLICATION_JSON)
             .header(CONTENT_TYPE, header::X_WWW_FORM_URLENCODED);
 
+        if let Some(user_agent) = config.user_agent() {
```

Comment:
Please add a suggestion to the documentation on the trait for custom implementations: `Implementations *should* use the user_agent to...`